torch/csrc/api/include/torch/data/ctf/ctf_parser.h

```diff
@@ -0,0 +1,931 @@
+#pragma once
+
+#include <torch/data/ctf/reader.h>
+#include <torch/data/ctf/utils.h>
+
+#include <stdint.h>
+#include <algorithm>
+#include <cassert>
+#include <climits>
+#include <iostream>
+#ifdef CTF_DEBUG
```

Comment:
Nit: I think pyTorch uses _DEBUG to distinguish debug build.

Reply:
Pytorch doesnt seem to have a standard cpp construct for this, but I found this:


#ifdef DEBUG_LOG
#define COLOR "\033[31;1m"
#define RESET "\033[0m"
#define __DEBUG(msg, ...) fprintf(stderr, COLOR msg "%c" RESET, __VA_ARGS__);
#define DEBUG(...) __DEBUG(__VA_ARGS__, '\n')
#else
#define DEBUG(...) (void)0
#endif`
`
at `./lib/libshm/manager.cpp`